common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -47,7 +47,7 @@
   static {
     boolean _unaligned;
     String arch = System.getProperty("os.arch", "");
-    if (arch.equals("ppc64le") || arch.equals("ppc64")) {
+    if (arch.equals("ppc64le") || arch.equals("ppc64") || arch.matches("s390x")) {
```

Comment:
`arch.equals()`?

Reply:
Added a commit with the change `arch.equals("s390x")`, as it is on master branch.